service/controller/defaultnamespace/defaultnamespacetest/test_methods.go

```diff
@@ -26,6 +26,29 @@ func ServiceAccountsShouldEqual(t *testing.T, expected []*corev1.ServiceAccount,
 	}
 }
 
+func ServiceAccountsShouldEqualDeep(t *testing.T, expected []*corev1.ServiceAccount, actual []corev1.ServiceAccount) {
```

Comment:
The main reason is that the test was added before the needsUpdate method was updated to its final version.
However, it still makes sense to me to have the logic in the tests separated from the main code, so that the code does not get checked by the same code.